openhands/integrations/gitlab/gitlab_service.py

```diff
@@ -79,4 +79,30 @@ def provider(self) -> str:
     'OPENHANDS_GITLAB_SERVICE_CLS',
     'openhands.integrations.gitlab.gitlab_service.GitLabService',
 )
-GitLabServiceImpl = get_impl(GitLabService, gitlab_service_cls)
+
+# Lazy loading to avoid circular imports
+_gitlab_service_impl = None
+
+
+def get_gitlab_service_impl():
+    """Get the GitLab service implementation with lazy loading."""
+    global _gitlab_service_impl
+    if _gitlab_service_impl is None:
+        _gitlab_service_impl = get_impl(GitLabService, gitlab_service_cls)
```

Comment:
sucks we have to do this but can't be worked around i suppose, thanks for digging into it!